src/opm/parser/eclipse/EclipseState/Schedule/UDQContext.cpp

```diff
@@ -0,0 +1,60 @@
+/*
+  Copyright 2019 Statoil ASA.
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <opm/parser/eclipse/EclipseState/Schedule/TimeMap.hpp>
+#include <opm/parser/eclipse/EclipseState/Schedule/UDQContext.hpp>
+#include <opm/parser/eclipse/EclipseState/Schedule/SummaryState.hpp>
+
+
+namespace Opm {
+
+    UDQContext::UDQContext(const SummaryState& summary_state) :
+        summary_state(summary_state)
+    {
+        for (const auto& pair : TimeMap::eclipseMonthIndices())
+            this->add(pair.first, pair.second);
+
+        /*
+          Simulator performance keywords which are expected to be available for
+          UDQ keywords; probably better to guarantee that they are present in
+          the underlying summary state object.
+        */
```

Comment:
Just out of curiosity, would it be possible to get this effect by reusing/extending the mechanism that's already in place for guaranteeing that certain [summary](https://github.com/OPM/opm-common/blob/cf0b7db5349cbbd86afa45d22dd9da2833ec14d1/src/opm/output/eclipse/Summary.cpp#L69-L70) vectors are always available in the `SummaryState` for the purpose of simulation restart?

Reply:
I don't have the full overview of the UDQ functionality, it is a bit discover as you go -  but with that disclaimer I would answer no to your question.

In the case of restart we know how to calculate the keyword, it has just not been asked for, in the UDQ case it is exactly opposite.

Reply:
> with that disclaimer I would answer no to your question.

Okay, thanks a lot for the background.